docs/specs/tsd-001-deprecation.md

```diff
@@ -0,0 +1,181 @@
+# Plugin and Plugin Option Deprecation
+
+## Objective
+
+Specifies the process of deprecating and removing plugins, plugin settings
+including values of those settings or features.
+
+## Keywords
+
+procedure, framework, all plugins
+
+## Overview
+
+Over time the number of plugins, plugin options and plugin features grow and
+some of those plugins or options are either not relevant anymore, have been
+superseded or subsumed by other plugins or options. To be able to remove those,
+this specification defines a process to deprecate plugins, plugin options and
+plugin features including a timeline and minimal time-frames. Additionally, the
+specification defines a framework to annotate deprecations in the code and
+inform users about such deprecations.
+
+## User experience
+
+In the deprecation phase a warning will be shown at Telegraf startup with the
+following content
+
+```text
+Plugin "inputs.logparser" deprecated since version 1.15.0 and will be removed in 2.0.0: use 'inputs.tail' with 'grok' data format instead
+```
+
+Similar warnings will be shown when removing plugin options or option values.
+This provides users with time to replace the deprecated plugin in their
+configuration file.
+
+After the shown release (`v2.0.0` in this case) the warning will be promoted
+to an error preventing Telegraf from starting. The user now has to adapt the
+configuration file to start Telegraf.
+
+## Deprecation Process
+
+After reaching an agreement in the issue, you can start the deprecation process
+by following the steps below.
+
+### File issue
+
+In the filed issue you should outline which plugin, plugin option or feature
+you want to deprecate and *why*! Determine in which version the plugin should
+be removed.
+Consider moving the plugin to the Influx community repository as an external
+plugin if there is no straight-forward replacement.
+
+Try to reach an agreement in the issue before continuing and get a sign off
+from the maintainers!
+
+### Submit deprecation pull-request
+
+Send a pull request adding deprecation information to the code and update the
+plugin's `README.md` file. Depending on what you want to deprecate this
+comprises different locations and steps as detailed below.
+
+Once the deprecation pull-request is merged and Telegraf is released, we have
+to wait for the targeted Telegraf version for actually removing the code.
+
+You should consider writing a blog-post to make users aware of the scheduled
+removal of plugins, plugin options or option values.
```

Comment:
You? Or the maintainers? Up to this point it reads like an external user would do all of this, but I would not expect an external user to write the blog post for this.

I also don't think we would do a blog post for every removal. Instead we would do a big post maybe one release before? or at the release we remove things in?

Reply:
This is a copy and past from the original document. So should I remove this paragraph?

Reply:
Yeah let's remove